### ISSUE TITLE
Support Typescript >= 3.5.0

### ISSUE THREAD

**Issue by Svish** [typescript-eslint/typescript-eslint#620]
**What code were you trying to parse?**

```ts
// Any code
```

**What did you expect to happen?**

No warnings

**What actually happened?**

Got warnings, which causes eslint to fail. Although sometimes it doesn't fail (i.e. I get no warning), so not sure what's up with that... Maybe there's some sort of "if in a hurry, fail with warning"-switch where ever `typescript-estree` decides to output this warning. 🤔😛

Either way, would be great if it supported this latest Typescript minor version.

```
 =============

 WARNING: You are currently running a version of TypeScript which is not officially supported by typescript-estree.

 You may find that it works just fine, or you may not.

 SUPPORTED TYPESCRIPT VERSIONS: >=3.2.1 <3.5.0

 YOUR TYPESCRIPT VERSION: 3.5.1

 Please only submit bug reports when using the officially supported version.

 =============
```

**Versions**

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/typescript-estree` | `1.10.2` |
| `TypeScript`                           | `3.5.1` |
| `node`                                 | `11.4.0` |
| `npm`                                  | `6.7.0` |

**Comment by JamesHenry** [comment]
I think the only explanation for this is that you have a different version of typescript-estree installed to the one you have specified.

1.10.2 is not capable of producing this message, you can see the published code here: https://unpkg.com/@typescript-eslint/typescript-estree@1.10.2/dist/parser.js

```const SUPPORTED_TYPESCRIPT_VERSIONS = '>=3.2.1 <3.6.0';```

**Comment by Svish** [comment]
Hm, you're right. Ran `npm list` again, and turns out there's 4 of them there. two `1.10.2`, and two `1.6.0` from `react-scripts` / `@babel/core`. So guess it might be the last ones being used... I'll go see if there's any updating going on over there instead. 🙂

But, maybe would be an idea to output the version in the error message to make it clear where it's coming from? For example:

>  WARNING: You are currently running a version of TypeScript which is not officially supported by typescript-estree **version 1.6.0**.
